components/DatePicker/DayOffPicker.js

```diff
@@ -0,0 +1,70 @@
+import { useState } from "react";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+
+export default function DayOffPicker({ daysOff, onDateChange }) {
+  const today = new Date();
+  const nextMonthLastDate = new Date(
+    today.getFullYear(),
+    today.getMonth() + 2,
+    0
+  );
+  const [selectedDates, setSelectedDates] = useState([]);
+  const [isOpen, setIsOpen] = useState(false);
+
+  //checks if the date has already been selected.
+  //If so, it removes the date from the list. And can not selecte more than 5 dates.
+  function handleDateChange(date) {
+    let newDates;
+    if (selectedDates.includes(date)) {
+      newDates = selectedDates.filter((d) => d !== date);
+    } else if (selectedDates.length < 5) {
```

Comment:
But it allows already for a maximum of 5 dates. Using selectedDates.length < 5 ensures that a date is added only if there are currently fewer than 5 dates selected. Changing it to  <= 5 would allow for a sixth date to be added. 🧐

Reply:
You're right, my logic was off.